Content.Shared/Hands/Components/SharedHandsComponent.cs

```diff
@@ -445,11 +445,14 @@ private EntityCoordinates GetFinalDropCoordinates(EntityCoordinates targetCoords
             var dropLength = EntitySystem.Get<SharedInteractionSystem>().UnobstructedDistance(origin, other, ignoredEnt: Owner);
             dropLength = MathF.Min(dropLength, SharedInteractionSystem.InteractionRange);
 
-            var dropVector = origin.Position;
-            if (dropLength != 0)
-                dropVector += (other.Position - origin.Position).Normalized * dropLength;
+            var dropVector = origin.Position - other.Position;
+            var diff = dropVector.Length - dropLength;
 
-            return targetCoords.WithPosition(dropVector);
+            // If we come up short then offset the drop location.
+            if (diff > 0)
+                return targetCoords.Offset(dropVector.Normalized * diff);
```

Comment:
shouldnt this be
```suggestion
                return targetCoords.Offset(dropVector.Normalized * dropLength);
```

Reply:
dropLength is unbounded so if you click further it can potentially drop further. Speaking of I'll probably just pass in teh capped range for unobstructeddistance for muh performance.